gluonnlp/data/transforms.py

```diff
@@ -326,3 +330,179 @@ def __call__(self, sample, return_str=False):
             List of words or detokenized text
         """
         return self._detokenizer.detokenize(sample, return_str=return_str)
+
+
+class JiebaTokenizer(object):
+    r"""Apply the jieba Tokenizer.
+
+    Users of this class are required to `install jieba <https://github.com/fxsjy/jieba>`_
+
+    Parameters
+    ----------
+    lang : str
+        The language to tokenize. Default is "zh", i.e, Chinese.
+
+    Examples
+    --------
+    >>> tokenizer = JiebaTokenizer()
+    >>> tokenizer(u"我来到北京清华大学")
+    ['我',
+     '来到',
+     '北京',
+     '清华大学']
+    >>> tokenizer(u"小明硕士毕业于中国科学院计算所，后在日本京都大学深造")
+    ['小明',
+     '硕士',
+     '毕业',
+     '于',
+     '中国科学院',
+     '计算所',
+     '，',
+     '后',
+     '在',
+     '日本京都大学',
+     '深造']
+
+    """
+    def __init__(self):
+        try:
+            import jieba
+        except ImportError:
+            raise ImportError('jieba is not installed. You must install jieba in order to use the '
+                              'JiebaTokenizer. You can refer to the official installation guide '
+                              'in https://github.com/fxsjy/jieba')
+        self._tokenizer = jieba
+
+    def __call__(self, sample):
+        """
+
+        Parameters
+        ----------
+        sample: str
+            The Chinese sentence to tokenize. Better not to input sentence in other languages
+            since this class is mainly used for Chinese Word Segmentation.
+
+        Returns
+        -------
+        ret : list of strs
+            List of tokens
+        """
+        # we use default cutting mode provided by jieba, i.e., accurate mode
+        return [tok for tok in self._tokenizer.cut(sample) if tok != ' ' and tok != '']
+
+
+class NLTKStanfordSegmenter(object):
+    r"""Apply the Stanford Chinese Word Segmenter implemented in NLTK.
+
+    Users of this class are required to install Java, NLTK and download Stanford Word Segmenter
+
+    Parameters
+    ----------
+    path_to_jar : str, default ''
+        Path to stanford-segmenter.jar.
+
+    path_to_slf4j : str, default ''
+        Path to slf4j-api.jar.
+
+    path_to_dict : str, default ''
+        Path to external dictionary dict-chris6.ser.gz.
+
+    path_to_model : str, default ''
+        Path to pre-trained segmentation model.
+
+    path_to_sihan_corpora_dict : str, default ''
```

Comment:
instead of having 5 separate paths, I'd suggest having one root path in the argument instead.